netbox/dcim/forms.py

```diff
@@ -74,6 +74,17 @@ def clean(self):
         elif self.cleaned_data['mode'] == IFACE_MODE_TAGGED_ALL:
             self.cleaned_data['tagged_vlans'] = []
 
+        # Validate tagged VLANs; must be a global VLAN or in the same site
+        else:
```

Comment:
```suggestion
        elif self.cleaned_data['mode'] == IFACE_MODE_TAGGED:
```
No need to check for this on access ports with no tagged VLANs.